### DISCUSSION TITLE
testing en setup (ts-jest, typegoose, typegraphql)

### DISCUSSION THREAD

**Discussion by Sed9yDataBank** [typegoose/typegoose#584] [Q&A] [answered]
**I am using: "mongodb-memory-server": "^7.3.4"**
Is there a guide on how to setup testing env for jest nd typegoose? I know it's different after the new <7.0.0 mongodb-memory-server release.

**Comment by hasezoey** [comment, marked as answer]
its not really different in MMS 7.0.0, it just changed syntax to make it more user friendly, if you want to look how typegoose does it:
- [`globalSetup.ts`](https://github.com/typegoose/typegoose/blob/master/test/globalSetup.ts)
- [`globalTeardown.ts`](https://github.com/typegoose/typegoose/blob/master/test/globalTeardown.ts)
- [`setupFile.ts`](https://github.com/typegoose/typegoose/blob/master/test/setupFile.ts)
- [`utils/connect.ts`](https://github.com/typegoose/typegoose/blob/master/test/utils/connect.ts)
- [`jest.config.json`](https://github.com/typegoose/typegoose/blob/master/jest.config.json#L28-L33)

Note: typegoose uses a config (custom) to change between in-memory (MMS) and a other instance when further debugging is needed

[MMS migration guide to v7.0](https://nodkz.github.io/mongodb-memory-server/docs/guides/migrate7)

**Reply by Sed9yDataBank** [reply]
yes, I have noticed the config file. Can I ditch using it? since I am going to do basic unit testing.

**Reply by hasezoey** [reply]
if you dont want to switch between external and MMS, then yes, you can remove all related code, but if you want to inspect documents / the database afterwards, you either have to manually switch the code or make it never exit
